pkg/monitor/event.go

```diff
@@ -20,6 +19,9 @@ func startEventMonitoring(ctx context.Context, m Recorder, client kubernetes.Int
 	reMatchFirstQuote := regexp.MustCompile(`"([^"]+)"( in (\d+(\.\d+)?(s|ms)$))?`)
 
 	go func() {
+		// Track our last observed resource version from each event, used to re-establish the watch
+		// when it's requested rv gets too old for the server. (which may happen when apiservers cycle)
+		var rv string
```

Comment:
how does this code handle the case where the resource version it wishes to watch from has expired and it has to re-list from the source to get a new resource version and then diff the content of the events it now sees?